qiskit_experiments/calibration/calibrations.py

```diff
@@ -0,0 +1,824 @@
+        recursive behaviour is needed to handle Call instructions as the name of
+        the called instruction defines the scope of the parameter. Each time a Call
+        is found _assign recurses on the channel-assigned subroutine of the Call
+        instruction and the qubits that are in said subroutine. This requires a
+        careful extraction of the qubits from the subroutine and in the appropriate
+        order. Next, the parameters are identified and assigned. This is needed to
+        handle situations where the same parameterized schedule is called but on
+        different channels. For example,
+
+        .. code-block:: python
+
+            ch0 = Parameter("ch0")
+            ch1 = Parameter("ch1")
+
+            with pulse.build(name="xp") as xp:
+                pulse.play(Gaussian(duration, amp, sigma), DriveChannel(ch0))
+
+            with pulse.build(name="xt_xp") as xt:
+                pulse.call(xp)
+                pulse.call(xp, value_dict={ch0: ch1})
+
+        Here, we define the xp schedule for all qubits as a Gaussian. Next, we define a
+        schedule where both xp schedules are called simultaneously on different channels.
+
+        Args:
+            schedule: The schedule with assigned channel indices for which we wish to
+                assign values to non-channel parameters.
+            qubits: The qubits for which to get the schedule.
+            free_params: The parameters that are to be left free. See get_schedules for details.
+            group: The calibration group of the parameters.
+            cutoff_date: Retrieve the most recent parameter up until the cutoff date. Parameters
+                generated after the cutoff date will be ignored. If the cutoff_date is None then
+                all parameters are considered. This allows users to discard more recent values that
+                may be erroneous.
+
+        Returns:
+            ret_schedule: The schedule with assigned parameters.
+
+        Raises:
+            CalibrationError:
+                - If a channel has not been assigned.
+                - If a parameter that is needed does not have a value.
+        """
+        # 1) Restrict the given qubits to those in the given schedule.
+        qubit_set = set()
+        for chan in schedule.channels:
+            if isinstance(chan.index, ParameterExpression):
+                raise (
+                    CalibrationError(
+                        f"All parametric channels must be assigned before searching for "
+                        f"non-channel parameters. {chan} is parametric."
+                    )
+                )
+            if isinstance(chan, (DriveChannel, MeasureChannel)):
+                qubit_set.add(chan.index)
+
+            if isinstance(chan, ControlChannel):
+                for qubit in self._controls_config_r[chan]:
+                    qubit_set.add(qubit)
+
+        qubits_ = tuple(qubit for qubit in qubits if qubit in qubit_set)
```

Comment:
Can we find schedule that only has a control channel? l.625 adds qubits from the reverse map, but this is always a two qubit tuple in our model. However, this line decomposes the input `qubits`. So the channels say `qubit_set = {(0, 1)}` and input `qubits` say we need `(0, 1)`, this line yields `qubits_ = (, )` because neither 0 nor 1 is contained in the `qubit_set`.